### PR TITLE
sharedLinks struct should be exported

### LINKED ISSUE #1079: slackevents.sharedLinks struct is unexported
### What happened

I'm writing a test for my shared link unfurler, and the `LinkSharedEvent` struct has a field `Links` which is an array of `sharedLinks`.

https://github.com/slack-go/slack/blob/9de73cee5f30be540fd051e94c3753c6d89ea905/slackevents/inner_events.go#L185-L188

### Expected behavior

The `sharedLinks` type should be exported.

#### reproducible code

Try to create a `LinkSharedEvent` with a list of `Links`:
```
	e := slackevents.LinkSharedEvent{
		Links:            []slackevents.sharedLinks{},
	}
```

### Versions
- Go: 1.18
- slack-go/slack: v0.10.2

### PR DESCRIPTION
**Pull request by pmenglund**
Fixes #1079 - breaks API compatibility as a struct is exported, but I assume it was not intended to be unexported.

### CODE REVIEW

File: slackevents/inner_events_test.go [hunk 1]
```diff
@@ -1,7 +1,8 @@
-package slackevents
+package slackevents_test
```
Comment (kanata2) [inline slackevents/inner_events_test.go]: Why did you change package name?

**Review by kanata2 — COMMENTED** [review]
Thanks for your PR!

File: slackevents/inner_events_test.go (hunk 1 above)
Comment (pmenglund) [reply slackevents/inner_events_test.go]: To ensure it is tested using the exported API

Comment (kanata2) [reply slackevents/inner_events_test.go]: I see.
I think we don't need to change the package name for these tests, so let's leave it as it is 🙏

Comment (pmenglund) [reply slackevents/inner_events_test.go]: reverted

Comment (kanata2) [reply slackevents/inner_events_test.go]: Thanks.

**Review by kanata2 — APPROVED** [review]
LGTM! Thanks.

Actually, I think this changes don't break user's applications by updating slack-go, so maybe okay to release in the next patch version.
